### ISSUE TITLE
BlenderBIM reoccurring Line 109 error

### ISSUE THREAD

**Issue by JonnyManako** [IfcOpenShell/IfcOpenShell#4123]
Hi all, I'm brand new to the GIT community to apologies in advanced as I stumble my way through this.

I'm having a few reoccurring errors when using BlenderBIM the easiest to reproduce being the following;

Set up a new IFC project, create a slab, tab to enter edit mode then this error pops up, (the slab DOES enter edit mode and I can use it successfully, however as I'm having other buggy behavior such as drawings randomly corrupting etc and BB not interacting with Inkscape as shown in various tutorials, I thought it would be worth checking that I haven't done something silly with my set up Blender/BB set up.

Thanks in advance!

```
Python: Traceback (most recent call last):
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\geometry\operator.py", line 1615, in invoke
    return IfcStore.execute_ifc_operator(self, context, is_invoke=True)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\ifc.py", line 334, in execute_ifc_operator
    result = getattr(operator, "_invoke")(context, None)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\geometry\operator.py", line 1620, in _invoke
    return self.execute(context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\geometry\operator.py", line 1505, in execute
    return IfcStore.execute_ifc_operator(self, context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\ifc.py", line 336, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\geometry\operator.py", line 1569, in _execute
    operator()
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\geometry\operator.py", line 1554, in <lambda>
    operator = lambda: bpy.ops.bim.hotkey(hotkey="S_E")
  File "C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py", line 109, in __call__
    ret = _op_call(self.idname_py(), kw)
RuntimeError: Error: Python: Traceback (most recent call last):
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\tool\ifc.py", line 201, in execute
    IfcStore.execute_ifc_operator(self, context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\ifc.py", line 336, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\model\workspace.py", line 542, in _execute
    getattr(self, f"hotkey_{self.hotkey}")()
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\model\workspace.py", line 631, in hotkey_S_E
    bpy.ops.bim.enable_editing_extrusion_profile()
  File "C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py", line 109, in __call__
    ret = _op_call(self.idname_py(), kw)
RuntimeError: Error: Python: Traceback (most recent call last):
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\tool\ifc.py", line 201, in execute
    IfcStore.execute_ifc_operator(self, context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\ifc.py", line 336, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
  File "C:\Users\Jonny\AppData\Roaming\Blender Foundation\Blender\4.0\scripts\addons\blenderbim\bim\module\model\slab.py", line 628, in _execute
    bpy.ops.wm.tool_set_by_id(tool.Blender.get_viewport_context(), name="bim.cad_tool")
  File "C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py", line 106, in __call__
    C_exec, C_undo = _BPyOpsSubModOp._parse_args(args)
  File "C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py", line 60, in _parse_args
    raise ValueError("1-2 args execution context is supported")
ValueError: 1-2 args execution context is supported
Location: C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py:109
Location: C:\Program Files\Blender Foundation\Blender 4.0\4.0\scripts\modules\bpy\ops.py:109
```

**Comment by theoryshaw** [comment]
Welcome.  Can you do a videocapture of the steps you did to get your first error?

If relatively new to BB, https://community.osarch.org/ is great for asking newbie questions.

**Comment by JonnyManako** [comment]
Ah yes! Thank you.
I have been on there in the recent past, but wasn't sure if this or that was the best place for BB questions.

At any rate, I have recorded the slab error.

https://youtu.be/zd35qzRYF7s
Apologies for the audio quality and random cutoff point.

I should mention;
BB V0.0.231104-4e76f36
Blender 4.0.2
Fresh install and all set back to default settings etc.
the only plugin I have running is BB.

**Comment by theoryshaw** [comment]
Nice.. this seems like a legitimate bug. 

In the future, you can just upload .mp4's (and perhaps other video formats) here, instead of hassling with youtube upload.

**Comment by JonnyManako** [comment]
Thanks for the tips.
Also glad to hear this looks like a bug (and it's not just be doing something silly).

Thanks so much for your time! Really appreciate it.

**Comment by Andrej730** [comment]
@JonnyManako hello! It was fixed in 48b237ab5463c952081fef334116befe8e2e56ee, you can download the latest BlenderBIM version from the [Releases](https://github.com/IfcOpenShell/IfcOpenShell/releases/) page.

**Comment by JonnyManako** [comment]
Wow.... That would have taken Graphisoft a few months. Thanks so much to who ever put the time in to patch that! So impressed and grateful for your efforts. Downloaded and tested, seems to be working a treat.
